lib/Doctrine/ORM/Internal/Hydration/ScalarColumnHydrator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Internal\Hydration;
+
+/**
+ * Hydrator that produces one-dimensional array "graph" that is often (not always) interchangeable
+ * with the corresponding object graph for read-only access.
+ */
+class ScalarColumnHydrator extends AbstractHydrator
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function hydrateAllData()
+    {
+        if (!isset($this->_rsm->indexByMap['scalars'])) {
+            return $this->_stmt->fetchOne();
+        }
+
+        $result = $this->hydrateAssociativeData();
+
+        if (empty($result)) {
+            return $result;
+        }
+
+        $indexColumn = $this->_rsm->scalarMappings[$this->_rsm->indexByMap['scalars']];
+
+        $keys = array_keys(reset($result));
+
+        $column = isset($keys[1]) && $keys[0] === $indexColumn ? $keys[1] : $keys[0];
+
+        return array_column($result, $column, $indexColumn);
```

Comment:
I don't understand why this needs to be that complicated…

The `ScalarHydrator` does more than this hydrator, but its code is more simple: https://github.com/doctrine/orm/blob/1de28c2cab5cd7e3c9d781a8beff22e328190911/lib/Doctrine/ORM/Internal/Hydration/ScalarHydrator.php#L32-L52


Why is all this needed?

Reply:
https://gist.github.com/ostrolucky/f9f1e0b271357573fde55b7a2ba91a32

Based on this implementation, I am trying to cover situations with multiple columns selected and if Scalar Values provided.

```php
 $result = []; 
  
         while ($data = $this->_stmt->fetchOne()) { 
             $this->hydrateRowData($data, $result); 
         } 
  
         return $result; 
```

Is this enough ? @greg0ire 


```php
$result[] = $this->gatherScalarRowData($row); 
```

Also `fetchOne` gives non array values so `gatherScalarRowData` throws errors

Reply:
It introduces infinite loop problem